packages/api-explorer/src/Doc.jsx

```diff
@@ -70,69 +70,56 @@ class Doc extends React.Component {
     this.enableRequestBodyJsonEditor = false;
     this.hideResults = this.hideResults.bind(this);
 
-    this.oas = new Oas(this.props.oas, this.props.user);
-
     this.onChange = this.onChange.bind(this);
     this.onJsonChange = this.onJsonChange.bind(this);
     this.onModeChange = this.onModeChange.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
 
     this.operation = this.getOperation();
-    this.Params = createParams(this.oas, this.operation);
+    this.Params = createParams(this.props.oas, this.operation);
 
     this.resetForm = this.resetForm.bind(this);
 
     this.toggleAuth = this.toggleAuth.bind(this);
     this.waypointEntered = this.waypointEntered.bind(this);
-  }
-
-  componentDidMount() {
-    if (!this.shouldEnableRequestBodyJsonEditor()) {
-      return;
-    }
 
-    this.operation
-      .getRequestBodyExamples()
-      .then(examples => {
-        if (!Object.keys(examples).length) {
-          return;
-        }
+    // If we need to build out the request body JSON editor let's pull examples to fill it with.
+    if (this.shouldEnableRequestBodyJsonEditor()) {
+      const examples = this.operation.getRequestBodyExamples();
```

Comment:
How did this become synchronous? Where's that code?

Reply:
@domharrington That happened in https://github.com/readmeio/oas/pull/327